pandas/tests/indexes/interval/test_interval_new.py

```diff
@@ -21,45 +21,25 @@ def _compare_tuple_of_numpy_array(self, result, expected):
 
     @pytest.mark.parametrize("idx_side", ['right', 'left', 'both', 'neither'])
     @pytest.mark.parametrize("side", ['right', 'left', 'both', 'neither'])
-    def test_get_loc_interval(self, idx_side, side):
-
+    @pytest.mark.parametrize("bound", [
+        [0, 1], [1, 2], [2, 3], [3, 4], [0, 2], [2.5, 3], [-1, 4]], ids=str)
+    def test_get_loc_interval(self, idx_side, side, bound):
+        # TODO: add overlapping/non-monotonic/dupe index tests
         idx = IntervalIndex.from_tuples([(0, 1), (2, 3)], closed=idx_side)
 
-        for bound in [[0, 1], [1, 2], [2, 3], [3, 4],
-                      [0, 2], [2.5, 3], [-1, 4]]:
-            # if get_loc is supplied an interval, it should only search
-            # for exact matches, not overlaps or covers, else KeyError.
-            if idx_side == side:
-                if bound == [0, 1]:
-                    assert idx.get_loc(Interval(0, 1, closed=side)) == 0
-                elif bound == [2, 3]:
-                    assert idx.get_loc(Interval(2, 3, closed=side)) == 1
-                else:
-                    with pytest.raises(KeyError):
-                        idx.get_loc(Interval(*bound, closed=side))
+        # if get_loc is supplied an interval, it should only search
+        # for exact matches, not overlaps or covers, else KeyError.
+        if idx_side == side:
+            if bound == [0, 1]:
+                assert idx.get_loc(Interval(0, 1, closed=side)) == 0
+            elif bound == [2, 3]:
+                assert idx.get_loc(Interval(2, 3, closed=side)) == 1
```

Comment:
can the error tests be pulled out to a separate test

Reply:
Yes, with somewhat similar tradeoffs described above, but I don't think it'd be as bad.  This seems like it'd be more reasonable to split, but haven't fully scoped it out.